src/anticipate/decorators.py

```diff
@@ -85,51 +88,115 @@ def __call__(self, *args, **kwargs):
             except AdaptError as e:
                 errors = [e]
 
-            raise AnticipateTypeError('Return value %r does not match anticipated type %r' % (type(result), self.returns), errors=errors)
+            raise AnticipateErrors(
+                message='Return value %r does not match anticipated type %r'
+                    % (type(result), self.returns),
+                errors=errors)
         elif self.strict:
             if result is not None:
-                raise AnticipateTypeError('Return value %r does not match anticipated value of None' % (type(result),), errors=None)
+                raise AnticipateErrors(
+                    message='Return value %r does not match anticipated value '
+                    'of None' % type(result),
+                    errors=None)
             return None
         else:
             return result
 
+    def _get_adapter(self, to):
+        is_list = False
+        if isinstance(to, list):
+            # Value is a list of items matching the first element's type
+            if len(to) == 1:
+                is_list = True
+                to = to[0]
+            else:
+                raise TypeError('An anticipated list can only contain one type')
+
+        if not isinstance(to, type) and hasattr(to, 'adapt'):
+            # The to type is an object, not a class, use its adapt method
+            if is_list:
+                return partial(self._each, to.adapt)
+            else:
+                return to.adapt
+        elif is_list:
+            return partial(adapt_all, to_cls=to)
+        else:
+            return partial(adapt, to_cls=to)
+
+    def _each(self, func, iterable):
+        """
+        Runs `func` over each item in the iterable and returns a list.
+        Returns an empty list if iterable is `None`.
+        """
+        if value is None:
+            return []
+        return [func(obj) for obj in iterable]
 
 
 class anticipate(object):
     """
     A decorator that defines what a function/method expects.
 
-    :param returns: Defines the type of object this function is expected to return.
-                    Python basetypes as well as any object is supported. `None` means
-                    that nothing can be returned. Use a list (ex: `[MyClass]`) to denote
-                    that an iterable of `MyClass` will be returned.
-    :param params: A list of key/type pairs. Each key corresponds to a parameter of the wrapped
-                   function. The type is the type of object the function expects for that parameter.
+    Args:
+        returns (mixed): Defines the type of object this function is expected
+            to return. Python basetypes as well as any object is supported.
+            `None` means that nothing can be returned. Use a list
+            (ex: `[MyClass]`) to denote that a list of `MyClass` will be
+            returned.
+        params (list): A list of key/type pairs. Each key corresponds to a
```

Comment:
Hm. It is a `dict`. I just updated the format without paying attention to the message. Will fix.

 It's `params` as apposed to `kwargs` because it describes the parameters of the function you are wrapping. It's not arbitrary dict of options like `kwargs` is generally used for.